src/test/java/com/list/todo/it/UserTest.java

```diff
@@ -1,139 +1,451 @@
+        @Override
+        public boolean supportsParameter(MethodParameter parameter) {
+            return parameter.getParameterType().isAssignableFrom(UserPrincipal.class);
+        }
+
+        @Override
+        public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer,
+                                      NativeWebRequest webRequest, WebDataBinderFactory binderFactory) throws Exception {
+            UserPrincipal userPrincipal = new UserPrincipal();
+            userPrincipal.setId(currentUserId);
+            userPrincipal.setUsername("username");
+            userPrincipal.setPassword(new BCryptPasswordEncoder().encode("password"));
+
+            return userPrincipal;
+        }
+    };
+
+    private HandlerMethodArgumentResolver putPageable = new HandlerMethodArgumentResolver() {
+        @Override
+        public boolean supportsParameter(
+                MethodParameter parameter) {
+            if (parameter.getParameterType().equals(
+                    Pageable.class)) {
+                return true;
+            }
+            return false;
+        }
+
+        @Override
+        public Object resolveArgument(
+                MethodParameter parameter,
+                ModelAndViewContainer mavContainer,
+                NativeWebRequest webRequest,
+                WebDataBinderFactory binderFactory) throws Exception {
+
+            return new PageRequest(0, 50);
+        }
+    };
+
+    @Before
+    public void before() {
+        mockMvc = MockMvcBuilders
+                .standaloneSetup(userController)
+                .setCustomArgumentResolvers(putAuthenticationPrincipal, putPageable)
+                .build();
+    }
+
 
     @Test
-    public void userInfo() throws Exception {
+    public void userInfo_OnExistentUser_ReturnsAnObjectOfUserSummary() throws Exception {
+        //arrange
+        when(userService.getUserInfo(any(UserPrincipal.class))).thenReturn(new UserSummary("stepanich", "Stepan Matveev", "stepa.matv72@gmail.com", null));
+
+        //act, assert
         this.mockMvc.perform(get("/api/users/me"))
                 .andDo(print())
-                .andExpect(authenticated())
                 .andExpect(jsonPath("name").value("Stepan Matveev"))
                 .andExpect(jsonPath("username").value("stepanich"))
                 .andExpect(jsonPath("email").value("stepa.matv72@gmail.com"))
                 .andExpect(status().isOk());
-
     }
 
     @Test
-    public void serchUsersByUsername() throws Exception {
+    public void serchUsersByUsername_OnExistentUser_ReturnsASetOfStrings() throws Exception {
+        //arrange
+        Set usernames = new HashSet();
+        usernames.add("anna");
+        when(userService.searchUsersByPartOfUsername(anyString())).thenReturn(usernames);
+
+        //act, assert
         this.mockMvc.perform(get("/api/users/search?username={username}", "ann"))
```

Comment:
Looks like this test should actually fail. Don't understand how it works)) You pass "ann" instead of "anna" here.
Verify that appropriate service method was called

Reply:
I see, you haven't got rid of any() matcher

Reply:
It searches by the part of username :)

Reply:
Got rid of matchers in any suchlike situations, please